### PR TITLE
Add MaxDbStatementLength option with default of 4096

### LINKED ISSUE #369: Limit size of db statement in Elasticsearch
Db statements can be large and shouldn't be unbounded. It would be nice to add an option to limit their size.

### PR DESCRIPTION
**Pull request by ejsmith**
Fixes #369

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs [hunk 1]
```diff
@@ -285,7 +283,7 @@ private string ParseAndFormatRequest(Activity activity, string debugInformation)
                     body = Encoding.UTF8.GetString(stream.ToArray());
                 }
 
-                return $"{method} {url}\r\n{body}";
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.ElasticsearchClient/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs:288]: Does this change the expected contents of the tag `db.statement`? If yes, then that needs to be mentioned in the CHANGELOG as well.

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs [hunk 2]
```diff
@@ -40,6 +40,11 @@ public class ElasticsearchClientInstrumentationOptions
         /// </summary>
         public bool SetDbStatementForRequest { get; set; } = true;
 
+        /// <summary>
+        /// Gets or sets a max length allowed for the db.statement attribute. Default value: 4096.
+        /// </summary>
+        public int MaxDbStatementLength { get; set; } = 4096;
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs:46]: Even though there hasn't been a stable release of this package yet, tt would be nice to have the publicAPI analyzers turned on for this project to detect any breaking changes.

**Review by utpilla — APPROVED** [review]
Approved with suggestions.

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs (hunk 2 above)
Comment (ejsmith) [reply src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs:46]: Not sure how to do that.

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs (hunk 1 above)
Comment (ejsmith) [reply src/OpenTelemetry.Instrumentation.ElasticsearchClient/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs:288]: Updated

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,8 @@
 * Updated OTel SDK package version to 1.2.0
 * Update minimum full framework support to net462
 * Requests that get an HTTP status code of 404 are not marked as an error span status
+* Add MaxDbStatementLength option with default of 4096
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.ElasticsearchClient/CHANGELOG.md:8]: It's generally a good idea to mention the PR numbers associated with the changes in the CHANGELOG.

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.ElasticsearchClient/CHANGELOG.md:8]: You could look at the OTel repo for reference: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/CHANGELOG.md

File: src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs (hunk 2 above)
Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs:46]: You could refer to this doc for more information: https://github.com/dotnet/roslyn-analyzers#microsoftcodeanalysispublicapianalyzers

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.ElasticsearchClient/ElasticsearchClientInstrumentationOptions.cs:46]: @ejsmith I have created a PR #374 to add public API analyzer to ElasticSearchClient project. Please review it.

**Pull request merged**